eng/pipelines/templates/BuildAndTest.yml

```diff
@@ -89,23 +85,25 @@ steps:
 
     - task: CopyFiles@2
       inputs:
-        Contents: '${{ parameters.repoTestResultsPath }}/**/*.cobertura.xml'
-        TargetFolder: '${{ parameters.repoTestResultsPath }}/collected'
+        Contents: '${{ parameters.repoArtifactsPath }}/**/*.cobertura.xml'
+        TargetFolder: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
         flattenFolders: true
-      displayName: Gather coverage results (cobertura.xml)
+      displayName: Gather code coverage results
 
-    - script: $(Build.SourcesDirectory)/.dotnet/dotnet dotnet-coverage merge
-              ${{ parameters.repoTestResultsPath }}/collected/*.cobertura.xml
-              --output-format cobertura
-              --output ${{ parameters.repoTestResultsPath }}/$(Agent.Os)_$(Agent.JobName).cobertura.xml
-      displayName: Merge code coverage reports
+    - ${{ if eq(parameters.runAsPublic, 'true') }}:
+      - task: PublishPipelineArtifact@1
+        displayName: Publish coverage results (cobertura.xml)
+        inputs:
+          targetPath: '${{ parameters.repoArtifactsPath }}/CodeCoverage'
+          artifactName: '$(Agent.JobName)_CodeCoverageResults'
+          publishLocation: 'pipeline'
 
-    - task: PublishBuildArtifacts@1
-      inputs:
-        PathtoPublish: '${{ parameters.repoTestResultsPath }}/$(Agent.Os)_$(Agent.JobName).cobertura.xml'
-        PublishLocation: Container
-        ArtifactName: CodeCoverageResults
-      displayName: Publish coverage results (cobertura.xml)
+    - ${{ if ne(parameters.runAsPublic, 'true') }}:
+      - task: 1ES.PublishPipelineArtifact@1
```

Comment:
What is the difference between PublishBuildArtifact and PublishPipelineArtifact?

Reply:
I used it as per a suggestion in the Teams channel.

In summary, if you need to share build outputs within the same pipeline or across different pipelines, use the “Publish Build Artifact” task. [If you want a more efficient way to transfer files between jobs within the same pipeline, opt for the “Publish Pipeline Artifact” task](https://learn.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops).